PYME/LMVis/Extras/clusterAnalysis.py

```diff
@@ -93,6 +93,58 @@ def OnClumpDBSCAN(self, event=None):
 
             pipeline.selectDataSource(clumper.outputName)
 
+    def OnRipleys(self, event=None):
+        """
+        Run's Ripley's K or L on the current dataset.
+        """
+        from PYME.LMVis.renderers import HistogramRenderer, Histogram3DRenderer
+        from PYME.simulation import locify
+        from PYME.IO import tabular
+        from PYME.recipes.measurement import Ripleys
+        import matplotlib.pyplot as plt
+
+        pipeline = self.visFr.pipeline
+        recipe = pipeline.recipe
+
+        # Generate a histogram rendering of the dataset as a mask for Ripley's K
+        two_d = np.count_nonzero(pipeline.dataSources[pipeline.selectedDataSourceKey]['z']) == 0
+        renderer = HistogramRenderer(self.visFr, pipeline) if two_d else Histogram3DRenderer(self.visFr, pipeline)
+        imf = renderer.GenerateGUI()
+
+        # Simulate random uniform points
+        x, y, z = locify.locify(imf.image.data, imf.image.pixelSize, 1)
```

Comment:
Looking at how locify works, you almost certainly don't want to use locify with p=1 to generate your mask points. Am currently transferring the mask logic from here to the non-gui code and will fix as I do that.

Locify is likely to be good for doing the MC simulation for confidence intervals.